src/main/scala/fs2/aws/s3/s3.scala

```diff
@@ -18,32 +18,25 @@ package object s3 {
       s3Client: S3Client[F] = new S3Client[F] {})(implicit F: Effect[F]): fs2.Stream[F, Byte] = {
     def go(offset: Int)(implicit F: Effect[F]): fs2.Pull[F, Byte, Unit] =
       fs2.Pull
-        .acquire[F, Option[S3ObjectInputStream]]({
-          try s3Client
-            .getObjectContent(
-              new GetObjectRequest(bucket, key).withRange(offset, offset + chunkSize))
-            .map(Some.apply)
-          catch {
-            case _: AmazonS3Exception => F.pure(None)
-          }
-        })({
-          case Some(s) => F.delay(s.abort())
-          case None    => F.delay(() => ())
-        })
-        .flatMap({
-          case Some(s) =>
+        .acquire[F, Either[Throwable, S3ObjectInputStream]](s3Client.getObjectContent(
+          new GetObjectRequest(bucket, key).withRange(offset, offset + chunkSize))) {
+          case Left(_)  => F.delay(() => ())
+          case Right(s) => F.delay(s.abort())
+        }
+        .flatMap {
+          case Right(s) =>
```

Comment:
maybe here we can rename the variables, since now there is the InputStream and the S3ObjectInputStream and only one letter separates one from the other